### PR TITLE
Modify UI  and add hint

### PR DESCRIPTION
**Pull request by choucl**
- I have modified the UI to make the score board change its order with the highness of scores
- Add tile placement hint with "H" command

### CODE REVIEW

File: src/client.c [hunk 1]
```diff
@@ -67,6 +63,19 @@ int choose_tile_handler(int c, rcb_t *rcb)
         render_board_preview(rcb);
         break;
     }
+    case 'H':
+        shift_msg(msg, color);
+        snprintf(msg[6], MAX_LOG_LEN,
+                 "[Hint] You could place tile \"%c\" at (%x, %x)",
+                 TILE[gcb->hint.shape].alpha, gcb->hint.pos.y, gcb->hint.pos.x);
+        *color[6] = BLUE_PAIR;
+        render_message_log(msg, color);
```
Comment (yuyuranium) [inline src/client.c:72]: You forgot to break.

File: src/client.c [hunk 2]
```diff
@@ -141,6 +147,19 @@ int positioning_handler(int c, rcb_t *rcb, char *msg[7], int *color[7])
                  chosen->pos.y, chosen->pos.x);
         render_message_log(msg, color);
         break;
+    case 'H':
+        shift_msg(msg, color);
+        snprintf(msg[6], MAX_LOG_LEN,
+                 "[Hint] You could place tile \"%c\" at (%x, %x)",
+                 TILE[gcb->hint.shape].alpha, gcb->hint.pos.y, gcb->hint.pos.x);
+        *color[6] = BLUE_PAIR;
+        render_message_log(msg, color);
```
Comment (yuyuranium) [inline src/client.c:156]: Here also.

File: src/client.c (hunk 1 above)
Comment (choucl) [reply src/client.c:72]: True. I'll fix it

File: src/render.c [hunk 3]
```diff
@@ -219,31 +220,26 @@ int render_message_log(char *msg[7], int *color[7])
...
-        } else if (gcb->score[p] < gcb->score[!p]) {
-            attroff(COLOR_PAIR(RED_PAIR));
-        }
+    mvprintw(18, 61, "Player %d", gcb->turn);
+    if (gcb->score[0] > gcb->score[1]) {
+        mvprintw(23, 59, "Player 0: %2d", gcb->score[0]);
+        mvprintw(24, 59, "Player 1: %2d", gcb->score[1]);
+    } else {
+        mvprintw(23, 59, "Player 1: %2d", gcb->score[1]);
+        mvprintw(24, 59, "Player 0: %2d", gcb->score[0]);
```
Comment (yuyuranium) [inline src/render.c:242]: Goooood!

File: src/client.c [hunk 4]
```diff
@@ -67,6 +63,19 @@ int choose_tile_handler(int c, rcb_t *rcb)
...
+    default:
+        shift_msg(msg, color);
+        snprintf(msg[6], MAX_LOG_LEN, "[Warning] Invalid operation \"%c\"", c);
```
Comment (yuyuranium) [inline src/client.c:76]: Directly print `c` may break UI when pressing enter.
image

Comment (yuyuranium) [reply src/client.c:76]: Some other non-printable characters such as control characters or combinations of ctrl + some keys, for example, may also break UI.

**Comment by choucl** [comment]
I will manage to print only printable characters; otherwise just print "unrecognized input"

**Comment by yuyuranium** [comment]
What if do nothing instead of print "unrecognized input"?

**Comment by choucl** [comment]
That'll be fine too👍

**Review by yuyuranium — COMMENTED** [review]
LGTM! Thanks. I'll merge it.

**Review by yuyuranium — APPROVED** [review]

**Pull request merged**